File: JobApplicationMicroservice/src/main/java/com/licenta/jobapplicationmicroservice/persistence/document/JobApplication.java
package com.licenta.jobapplicationmicroservice.persistence.document;

import com.licenta.jobapplicationmicroservice.business.model.Job;
import com.licenta.jobapplicationmicroservice.business.model.Message;
import com.licenta.jobapplicationmicroservice.business.model.Review;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Builder
@Document(collection = "job_applications")
public class JobApplication {

    @Id
    private String id;

    private Long userId;

    private Job job;

    private JobApplicationStatus status;

    private List<Message> messageList;

    private Review review;

    private LocalDateTime updatedAt;
}

File: frontend/users/src/app/models/job_application.ts
import { Job } from "./job";
import { JobApplicationStatus } from "./job_application_status";
import { JobResponse } from "./job_response";
import { JobApplicationMessage } from "./job_application_message";
import { UserDetails } from "./user_details";
import { Review } from "./review";
import { FileData } from "./file_data";

export class JobApplication {
  id?: string;
  userId?: number;
  job?: JobResponse;
  updatedAt: Date;
  status?: JobApplicationStatus;
  userDetails?: UserDetails;
  jobDetails: Job;
  messageList: Set<JobApplicationMessage>;
  review: Review;
  formattedUpdatedAt: string;
  fileList?: FileData[];
}
